Parser/pegen.c

```diff
@@ -416,8 +431,13 @@ _PyPegen_raise_error_known_location(Parser *p, PyObject *errtype,
     }
 
     if (!error_line) {
```

Comment:
Could we add some comments here explaining a bit the same of your PR description: that is, what situation happens on each branch. I would also suggest to have some asserts on the different statuses of `p->tok->fp` and friends in case we are wrong and some branches trigger in some of the cases we don't expect.

Reply:
Added some comments. Let me know, if it needs more explanation.